classic/selection/FeatureModel.js

```diff
@@ -0,0 +1,238 @@
+     * @property {ol.Collection<ol.Feature>}
+     */
+    selectedFeatures: null,
+
+    /**
+     * The connected vector layer.
+     * @cfg {ol.layer.Vector}
+     * @property {ol.layer.Vector}
+     */
+    layer: null,
+
+    /**
+     * The OpenLayers map we work on
+     * @cfg {ol.Map}
+     */
+    map: null,
+
+    /**
+     * Set to true to create a click handler on the map selecting a clicked
+     * object in the #layer.
+     * @cfg {Boolean}
+     */
+    mapSelection: false,
+
+    /**
+     * The default style for the selected features.
+     * @cfg {ol.style.Style}
+     */
+    selectStyle: new ol.style.Style({
+        image: new ol.style.Circle({
+            radius: 6,
+            fill: new ol.style.Fill({
+                color: 'rgba(255,255,255,0.8)'
+            }),
+            stroke: new ol.style.Stroke({
+                color: 'darkblue',
+                width: 2
+            })
+        }),
+        fill: new ol.style.Fill({
+            color: 'rgba(255,255,255,0.8)'
+        }),
+        stroke: new ol.style.Stroke({
+            color: 'darkblue',
+            width: 2
+        })
+    }),
+
+    /**
+     * The attribute key to mark an OL feature as selected.
+     * @cfg {String}
+     * @property
+     * @readonly
+     */
+    selectedFeatureAttr: 'gx_selected',
+
+    /**
+     * Prepare several connected objects once the selection model is ready.
+     *
+     * @private
+     */
+    bindComponent: function() {
+        var me = this;
+
+        me.callParent(arguments);
+
+        me.selectedFeatures = new ol.Collection();
+
+        // detect a layer from the store if not passed in
+        if (!me.layer || !me.layer instanceof ol.layer.Vector) {
+            var store = me.getStore();
+            if (store && store.getLayer() &&
+                store.getLayer() instanceof ol.layer.Vector) {
+                me.layer = store.getLayer();
+            }
+        }
+
+        // bind several OL events
+        me.bindOlEvents();
+    },
+
+    /**
+     * Binds several events on the OL objects used in this class.
+     *
+     * @private
+     */
+    bindOlEvents: function() {
+        var me = this;
+
+        // change style of selected feature
+        me.selectedFeatures.on('add', me.onSelectFeatAdd, me);
+
+        // reset style of no more selected feature
+        me.selectedFeatures.on('remove', me.onSelectFeatRemove, me);
+
+        // create a map click listener for connected vector layer
+        if (me.mapSelection && me.layer && me.map) {
+            me.map.on('singleclick', me.onFeatureClick, me);
+        }
+    },
+
+    /**
+     * Handles 'add' event of #selectedFeatures.
+     * Ensures that added feature gets the #selectStyle.
+     *
+     * @private
+     * @param  {ol.Collection.Event} evt OL event object
+     */
+    onSelectFeatAdd: function(evt) {
+        var me = this;
+        var feat = evt.element;
+        if (feat) {
+            feat.setStyle(me.selectStyle);
```

Comment:
Should we save any existing style  on the feature, so that we ca later restore to that style e.g. in `onSelectFeatRemove`?